docs/domain/timestamp_tz/index.md

```diff
@@ -0,0 +1,74 @@
+
+When reading, usually converting to `LocalTimestampWithOffset` and returning only that to clients might be sufficient. When doing so, remember that `LocalTimestampWithOffset` cannot be evaluated on Postgres side, so avoid using this property in queries in contexts where its value would participate in some computation, e.g. sorting.
+
+On the other hand, when writing and when replicating saved inputs for readback, it might be desirable to have both properties in the contract. In such case, a DTO like
+
+```csharp
+public record TimestampTzDTO(DateTimeOffset Timestamp, string TimeZoneId)
+{
+    public class ErrorCodes
+    {
+        public const int InvalidTimeZoneId = 10001;
+    }
+}
+```
+
+should satisfy those needs, with mapping between the two being trivial.
+
+Because EF currently does not support including properties of nested objects in index definitions, indexing timestamps stored in `TimestampTz` requires building indexes manually, using [`MigrationBuilder.Sql` API](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.migrations.migrationbuilder.sql), for example:
+
+```csharp
+migrationBuilder.Sql(
+    """
+    create index "IX_Events_StartTime" on "public"."Events"
+    (("StartTime_UtcTimestamp" at time zone "StartTime_TimeZoneId"));
+    """
+);
+```
+
+## Limitations
+
+* This type does not account for the variable nature of time zone rules. When attempting to store a point of time when some future event is supposed to take place, keep in mind that it may be defined by the _local_ time in some time zone. If rules governing that time zone change, that local time may no longer map to the same instant. In such cases, consider additionally storing the local timestamp in a separate column. See the blog post linked at the top for details.
+* Support for different calendar systems is out of scope.
+
+## FAQ
+
+### Q: When to use `DateTime`, `DateTimeOffset` and `TimestampTz` in domain entities?
+
+A: In Postgres, use `DateTimeOffset` (mapped to `timestamp with time zone` column type) to represent an instant without associated time zone (e.g. various `CreatedAt` timestamps generated on the backend). Use `TimestampTz` to persist an instant in context of some time zone (e.g. in a system aggregating news, it could be a time of an event that took place in some location). Use `DateTime` mapped to `timestamp without time zone` column type _together with_ `TimestampTz` to represent the local time of some event, often one meant to start in the future, or for indexing purposes. Avoid using `DateTime` mapped to `timestamp with time zone` columns: it has no benefits over using `DateTimeOffset`.
```

Comment:
Won't +5 offset be problematic when saved in `timestamp with time zone`? How that will behave?